yarn-project/foundation/src/collection/object.ts

```diff
@@ -0,0 +1,19 @@
+/** Returns a new object with the same keys and where each value has been passed through the mapping function. */
+export function mapValues<K extends string | number | symbol, T, U>(
```

Comment:
I think this declaration is redundant?

Reply:
It isn't. If K is a union of strings (eg `'foo' | 'bar'`), then a `Record<K>` requires both `foo` and `bar` to be present, whereas the `Partial<Record<K>>` allows skipping values. Having both declarations means that the type checker should pick the `Record` one when possible, letting you check that the set of input keys matches the set of output keys.

Actually, as I'm writing this, I'm realizing there may be a better type signature for this. I can revisit in a future PR.